Application/gui_main.py

```diff
@@ -704,51 +740,43 @@ def init_menu(self) -> None:
-        solid_closure.triggered.connect(lambda: self.change_style('Closure', 'solid'))
-
-        closure_style_menu.addActions([dashed_closure, dashdot_closure, solid_closure])
+        for style, description in LINE_STYLES.items():
+            action = QAction(description, covid_style_menu)
+            action.setStatusTip(description)
+            action.triggered.connect(make_function(self.plot_canvas.update_lines,
+                                                   self.plot_canvas.covid_axes, style=style))
+            covid_style_menu.addAction(action)
+
+        for style, description in LINE_STYLES.items():
+            action = QAction(description, closure_style_menu)
+            action.setStatusTip(description)
+            action.triggered.connect(make_function(self.plot_canvas.update_lines,
+                                                   self.plot_canvas.closure_axes, style=style))
+            closure_style_menu.addAction(action)
 
         # Show marker toggle menu
-        marker_menu = self.settings_menu.addMenu('Show Marker')
-
-        covid_marker = QAction('COVID-19 Plot', self)
-        covid_marker.setCheckable(True)
-        covid_marker.setChecked(True)
-        covid_marker.setStatusTip('Show/Hide markers in the COVID-19 Cases plot')
-        covid_marker.triggered.connect(self.toggle_marker_covid)
-
-        closure_marker = QAction('Closure Plot', self)
-        closure_marker.setCheckable(True)
-        closure_marker.setChecked(True)
-        closure_marker.setStatusTip('Show/Hide markers in the School Closure Status plot')
-        closure_marker.triggered.connect(self.toggle_marker_closure)
-
-        marker_menu.addActions([covid_marker, closure_marker])
+        marker_menu = self.settings_menu.addMenu('Markers')
+
+        covid_marker_menu = marker_menu.addMenu('COVID-19 Cases Line Marker')
+        closure_marker_menu = marker_menu.addMenu('School Closure Status Line Marker')
+
+        for marker, info in LINE_MARKERS.items():
+            description, icon_name = info
+            action = QAction(description, covid_marker_menu)
+            action.setIcon(QIcon(f'resources/assets/markers/{icon_name}'))
+            action.setStatusTip(description)
+            action.triggered.connect(make_function(self.plot_canvas.update_lines,
+                                                   self.plot_canvas.covid_axes, marker=marker))
+            covid_marker_menu.addAction(action)
+
+        for marker, info in LINE_MARKERS.items():
+            description, icon_name = info
+            action = QAction(description, closure_marker_menu)
+            action.setIcon(QIcon(f'resources/assets/markers/{icon_name}'))
+            action.setStatusTip(description)
+            action.triggered.connect(make_function(self.plot_canvas.update_lines,
+                                                   self.plot_canvas.closure_axes, marker=marker))
+            closure_marker_menu.addAction(action)
```

Comment:
Do you want to make this thing into the loop ontop also, just like the previous fix I sugggested
Instead of action, call it maybe "action_covid" and "action_closure"